scvi/model/_scvi.py

```diff
@@ -127,7 +128,10 @@ def __init__(
             library_log_means, library_log_vars = _init_library_size(
                 self.adata_manager, n_batch
             )
-
+        [self.batch_0, self.batch_1, self.batch_2, self.batch_3, self.batch_4] = [torch.tensor(adata.X[adata.obs['_scvi_batch'] == 0]), torch.tensor(adata.X[adata.obs['_scvi_batch'] == 1]),
+                                                                                  torch.tensor(adata.X[adata.obs['_scvi_batch'] == 2]), torch.tensor(adata.X[adata.obs['_scvi_batch'] == 3]),
+                                                                                  torch.tensor(adata.X[adata.obs['_scvi_batch'] == 4])]
+        self.batches = [self.batch_0, self.batch_1, self.batch_2, self.batch_3, self.batch_4]
```

Comment:
Look into this tutorial for data handling strategy: https://docs.scvi-tools.org/en/stable/tutorials/notebooks/dev/data_tutorial.html

Reply:
Think of edge cases: what if there are more than 5 batches?